src/backend/core/api/serializers.py

```diff
@@ -226,9 +230,36 @@ def validate_file(self, file):
                 f"File type '{mime_type:s}' is not allowed. Allowed types are: {mime_types:s}"
             )
 
+        # Skip conversion and pass through SVG files
+        if mime_type == "image/svg+xml":
+            return file  # Simply pass SVG files through without modification
+
+        # Resize and compress other image formats
+        elif mime_type and mime_type.startswith("image/"):
+            original_extension = os.path.splitext(file.name)[1]
+            image = Image.open(file)
+
+            max_width, max_height = settings.DOCUMENT_IMAGE_MAX_DIMENSIONS
+            image.thumbnail((max_width, max_height))
```

Comment:
Why using the "thumbnail" method? Is it really a thumbnail we want? I'm sorry because I don't have the initial context, but from the first review I was guessing we only want to "reduce" big images, but not making thumbnails, am I right?

Also, I think, I would not try to guess mime type nor extension: I would directly make:

```python
try:
    image = Image.open(file)
except PIL.UnidentifiedImageError: 
    # not an image or a supported format
else:
    # ...
    image.resize(....
```

for sure you may try to call PIL only `if mime_type and mime_type.startswith("image/")` but anyway, I would let PIL say if it can manage the format or not (and actually PIL does not manage SVG so you would not have to make a specific exception for them).

Reply:
@qbey 

according to your thumbnail question:

1. Aspect Ratio Preservation: Unlike some resizing methods that might distort the image to fit the specified dimensions, .thumbnail() ensures that the aspect ratio of the image is preserved.

2. Efficiency: .thumbnail() is optimized for performance. It resizes the image in place, meaning that it modifies the original image object directly without creating a new one, which can be faster and consume less memory than other approaches. The outcome is the same, it has nothing to do with a small image size or so, it still will be saved at a maximum resolution of 1920x1080 as currently set with DOCUMENT_IMAGE_MAX_DIMENSIONS.

Using PIL.UnidentifiedImageError, is a nice idea. I will implement this right away and update the PR accordingly

Reply:
Please don't merge yet, this appears to have conflicts with recent changes from main